### PR TITLE
feat: support firefly wallet in solana

### PR DESCRIPTION
**Pull request by dudu0506**
https://mask.atlassian.net/browse/FW-2618

# Pull Request

## Related Issue

Closes FW-0000

## Checklist

-   [ ] I have tested these changes locally.
-   [ ] I have reviewed the code changes.
-   [ ] I have added unit tests if applicable.

### CODE REVIEW

File: src/app/connectors/ParticleSolanaWallet.ts [hunk 1]
```diff
@@ -0,0 +1,311 @@
...
+        if (wallet) {
+            wallet.off('disconnect', this._disconnected);
+            wallet.off('accountChanged', this._accountChanged);
+
+            this._wallet = null;
+            this._publicKey = null;
+
+            try {
+                await wallet.disconnect();
+            } catch (error: any) {
```
Comment (guanbinrui) [inline src/app/connectors/ParticleSolanaWallet.ts:191]:
```suggestion
            try {
                await wallet.disconnect();
                
                wallet.off('disconnect', this._disconnected);
                wallet.off('accountChanged', this._accountChanged);
    
                this._wallet = null;
                this._publicKey = null;

            } catch (error: any) {
```

File: src/components/SolanaWalletAdapterProvider.tsx [hunk 2]
```diff
@@ -11,10 +11,12 @@ import { WalletModalProvider } from '@solana/wallet-adapter-react-ui';
 import { WalletConnectWalletAdapter } from '@solana/wallet-adapter-walletconnect';
 import { type PropsWithChildren } from 'react';
 
+import { ParticleSolanaWalletAdapter } from '@/app/connectors/ParticleSolanaWallet.js';
 import { env } from '@/constants/env.js';
 import { SOLANA_WALLET_CACHE_KEY } from '@/constants/index.js';
 
 const wallets: Adapter[] = [
+    new ParticleSolanaWalletAdapter(),
     new PhantomWalletAdapter(),
```
Comment (guanbinrui) [inline src/components/SolanaWalletAdapterProvider.tsx:20]: Please ensure that FireflyWallet is the first item on the modal.

File: src/app/connectors/ParticleSolanaWallet.ts [hunk 3]
```diff
@@ -0,0 +1,311 @@
...
+    private _disconnected = () => {
+        const wallet = this._wallet;
+        if (wallet) {
+            wallet.off('disconnect', this._disconnected);
+            wallet.off('accountChanged', this._accountChanged);
+
+            this._wallet = null;
+            this._publicKey = null;
+
+            this.emit('error', new WalletDisconnectedError());
```
Comment (guanbinrui) [inline src/app/connectors/ParticleSolanaWallet.ts:290]: Why error?

File: src/app/connectors/ParticleSolanaWallet.ts [hunk 4]
```diff
@@ -0,0 +1,311 @@
...
+                );
+                if (!wallets.length) {
+                    console.error(`[particle solana] wallet not found`);
+                    throw new AuthenticationError('Wallet not found');
+                }
+
+                try {
+                    // update wallet state
+                    await wallet.connect();
+                } catch (error: any) {
```
Comment (guanbinrui) [inline src/app/connectors/ParticleSolanaWallet.ts:151]: Use unknown to replace any

File: src/components/SolanaWalletAdapterProvider.tsx (hunk 2 above)
Comment (dudu0506) [reply src/components/SolanaWalletAdapterProvider.tsx:20]: Will try edit source code to fix this

**Review by guanbinrui — APPROVED** [review]

**Pull request merged**